README.md

```diff
@@ -1,19 +1,126 @@
-      * CORs as a way to stop other domains from using your internal API
-      * Not going to cover auth in a major way in this unit
+## Lesson
+
+We've seen how to *consume* various APIs, now let's talk about
+*providing* one.
+
+The choice to build and provide an API boils down to one of two options:
+
+1. Your data is valuable to more than just your application, or your
+   application has valuable integration points for other
+applications to provide new and interesting services, so you build an *external* API and open it up to third-party developers.
+2. Your data is only valuable to your company, but you access it in
+   multiple ways (e.g. a web application, iOS application, and Android
+application all for your company), or you have a distributed/service architecture, or you're even just using a front-end framework with Rails as an API back-end, so you build an
+*internal* API to provide common access to all internally-managed
+clients.
+
+### External API Considerations
+
+External APIs are extremely valuable to the growth of the web. Because
+of external APIs you can watch your Uber driver approach, plot your
+course with Google maps, get a text when the driver is close, and pay
+without ever touching your wallet.
+
+Because an external API is intended to be used by a wide variety of
+developers, with whom you wouldn't have much direct contact, there are a
+lot of things to account for when building one.
+
+**Documentation:** An external API must be *extensively* documented
+   to support your developer community. This includes not only
+functional specifications, but code samples, sometimes in multiple
+languages (human AND computer!).
+
+Sometimes you'll want to provide tools to live-test the API, such as Foursquare does. You might even provide libraries (such as a gem) for the
+languages you support.
+
+**Versioning:** We saw this with the Foursquare API. Versioning is an
+   important consideration, and you have to make sure that you don't
+release code in such a way that it suddenly breaks applications that
+rely on your API.
+
+**Authentication and Security:** Allowing external access to your data is a
+   serious thing, so your external API needs robust access control and
+security. We came across this when consuming the Foursquare and Github
+APIs. We had to register our application, pass ID and secret pairs to
+the API, and sometimes use OAuth tokens on behalf of a user - all
+security implementations that need to be set up and maintained by the
+API provider.
+
+**Quality of Service:** Once you have third-party developers relying
+   on your API, there are expectations around quality, uptime, and response
+time.
+
+You also have to consider things like **rate limiting**, where you
+monitor access and only allow a certain number of API calls per
+application in a given timeframe.
+
+We won't be covering much of this in this unit, but these are all
```

Comment:
As a student, when I see the phrase "We won't be covering much of this...", my eyes kind of glaze over and I forget what I'm about to read before even reading it (if that makes any sense).

Maybe we can instead say something that emphasizes that a given implementation of rate limits, uptime, etc. will vary based on the requirements and constraints of the given application, "but these are all important considerations if you plan to make your API public"?